scipy/signal/_peak_finding.py

```diff
@@ -233,6 +234,733 @@ def argrelextrema(data, comparator, axis=0, order=1, mode='clip'):
     return np.where(results)
 
 
+def peak_prominences(x, peaks, wlen=None):
+    """
+    Calculate the prominence of each peak in a signal.
+
+    The prominence of a peak measures how much a peak stands out from the
+    surrounding baseline of the signal and is defined as the vertical
+    distance between the peak and its lowest contour line.
+
+    Parameters
+    ----------
+    x : sequence
+        A signal with peaks.
+    peaks : sequence
+        Indices of peaks in `x`.
+    wlen : number, optional
+        A window length in samples that limits the search for the lowest
+        contour line to a symmetric interval around the evaluated peak. If not
+        given the entire signal `x` is used. Use this parameter to speed up the
+        calculation significantly for large vectors (see Notes).
+
+    Returns
+    -------
+    prominences : ndarray
+        The calculated prominences for each peak in `peaks`.
+    left_bases, right_bases : ndarray
+        The peaks' bases as indices in `x` to the left and right of each peak.
+        The higher base of each pair is a peak's lowest contour line (see Notes
+        for a more details).
+
+    See Also
+    --------
+    find_peaks, peak_widths
+
+    Notes
+    -----
+    Strategy to compute a peak's prominence:
+
+    * Extend a horizontal line from the current peak to the left and right until
+      the line either reaches the window end (see `wlen`) or intersects the
+      signal again at the slope of a higher peak.
+    * On each side find the minimal signal value within the interval defined
+      above. These points are the peak's bases.
+    * The higher one of the two bases marks the peak's lowest contour line. The
+      prominence can then be calculated as the vertical difference between the
+      peaks height itself and its lowest contour line.
+
+    Searching for the peak's bases can be slow for large `x` because the full
+    signal needs to be evaluated for each peak. This evaluation area can be
+    limited with the parameter `wlen` which restricts the algorithm to a window
+    around the current peak and can shorten the calculation time significantly.
+    However this may stop the algorithm from finding the true global contour
+    line if the peak's bases are outside this window. Instead a higher contour
+    line is found within the restricted window leading to a smaller calculated
+    prominence. In practice this is only relevant for the largest set of peaks
+    in vector. This behavior may even be used intentionally to calculate "local"
+    prominences.
+
+    .. versionadded:: 1.1.0
+
+    References
+    ----------
+    .. [1] Wikipedia Article for Topograhpic Prominence:
+       https://en.wikipedia.org/wiki/Topographic_prominence
```

Comment:
Is there a more authoritative reference for this (journal article)?

Reply:
Also, typo in Topographic on the line above

Reply:
I don't have a satisfying response as I come up empty handed when searching for papers. The stuff that I do find is in line with the linked article and deals with topographic analysis of peaks and mountains (nothing DSP related). Of course I could use some of the links on that Wikipedia article itself but I think that wouldn't improve the situation.

Honestly I first used Matlab's explanation as a guideline and later this Wikipedia article. Maybe something can be found later on but for now I'd like to leave this reference as it explains the term "prominence" (although 2D) in more detail than appropriate for this docstring.

Reply:
> Of course I could use some of the links on that Wikipedia article itself but I think that wouldn't improve the situation.

Yeah those don't look useful.

I had a look as well, it just doesn't seem to be a thing in DSP. So let's leave the Wikipedia reference